### DISCUSSION TITLE
default lancedb error

### DISCUSSION THREAD

**Discussion by yangboz** [safevideo/autollm#227] [Q&A]
it happens at python3.9, macosx13.5.2 , with pip install autollm, FastAPI example code ,after run it, it throw errors at following:
```
AutoQueryEngine.from_defaults(
  File "/Users/yangboz/anaconda3/envs/py39/lib/python3.9/site-packages/autollm/auto/query_engine.py", line 258, in from_defaults
    return create_query_engine(
  File "/Users/yangboz/anaconda3/envs/py39/lib/python3.9/site-packages/autollm/auto/query_engine.py", line 105, in create_query_engine
    vector_store_index = AutoVectorStoreIndex.from_defaults(
  File "/Users/yangboz/anaconda3/envs/py39/lib/python3.9/site-packages/autollm/auto/vector_store_index.py", line 84, in from_defaults
    lancedb_uri = AutoVectorStoreIndex._validate_and_setup_lancedb_uri(
  File "/Users/yangboz/anaconda3/envs/py39/lib/python3.9/site-packages/autollm/auto/vector_store_index.py", line 143, in _validate_and_setup_lancedb_uri
    raise ValueError(
ValueError: No existing database found at ./.lancedb. Please provide a valid lancedb uri.
Sentry is attempting to send 2 pending events
Waiting up to 2 seconds
Press Ctrl-C to quit
````

any ideas? thanks.

**Comment by yangboz** [comment]
even though I manually create a .lancedb  file  in current  project file system.

**Reply by yangboz** [reply]
```
from autollm import AutoQueryEngine
 import lancedb
 db = lancedb.connect("~/.lancedb") 
```

it comes from https://github.com/safevideo/autollm/#create-a-query-engine-in-seconds

**Reply by yangboz** [reply]
python3.9/pathlib.py", line 1323, in mkdir
    self._accessor.mkdir(self, mode)
FileExistsError: [Errno 17] File exists  maybe is it python mkdir to overwrite existed file  related ?

**Comment by AyushExel** [comment]
@SeeknnDestroy @fcakyon let me know if this error is related to lancedb's latest release in some way and I'll get it fixed

**Comment by fcakyon** [comment]
@yangboz can you please share an example code on your use case?

**Reply by yangboz** [reply]
so , I have tried rm the existed .lancedb file at first , then modified the code to 
import lancedb
db = lancedb.connect("./.lancedb"), it works, but after that the whole piece of code is ````query_engine=AutoQueryEngine.from_defaults(
# documents='...'
llm_model=llm_model,
llm_api_base=llm_api_base)
```
but after calling ```query_engine=AutoQueryEngine.from_defaults(
# documents='...'
llm_model=llm_model,
llm_api_base=llm_api_base)
 print("query_engine:",query_engine)

 response = query_engine.query(query_str)``` , it  throws errors as following,
```
  File "/Users/yangboz/anaconda3/envs/py39/lib/python3.9/site-packages/litellm/utils.py", line 6573, in exception_type
    raise APIConnectionError(
litellm.exceptions.APIConnectionError: The api_key client option must be set either by passing api_key to the client or by setting the OPENAI_API_KEY environment variable
''' I used
 llm_model = "ollama/llama2"
llm_api_base = "http://localhost:11434"
as default configs. 
any idea about this?  thanks.

**Reply by yangboz** [reply]
IMBO , ollama is local LLM , should't connect to OPEN_AI, right?

**Reply by fcakyon** [reply]
In your code, I see you did not set a local embedding model, so you still need openai-key for embedding generation.

**Reply by yangboz** [reply]
btw, I see the ollama provide a embbeding API here: [https://github.com/ollama/ollama/blob/main/docs/api.md#generate-embeddings]( https://github.com/ollama/ollama/blob/main/docs/api.md#generate-embeddings )
wondering without using  OPEN_AI_KEY purely in local environments is feasible?

**Comment by fcakyon** [comment]
[your provided code](https://github.com/safevideo/autollm/discussions/227#discussioncomment-8377053) and [the error you receive](https://github.com/safevideo/autollm/discussions/227#discussion-6181611) do not match. Are you sure you provided the full code you are having error?

In addition, `db = lancedb.connect("./.lancedb")` should be deleted from the code.
